src/endpoints/crosschain/ethereum/text_records.rs

```diff
@@ -0,0 +1,197 @@
+use anyhow::{Context, Result, anyhow};
+use reqwest::{Client, StatusCode};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use starknet::{
+    core::{
+        types::{BlockId, BlockTag, FieldElement, FunctionCall},
+        utils::{cairo_short_string_to_felt, parse_cairo_short_string},
+    },
+    macros::selector,
+    providers::{jsonrpc::HttpTransport, JsonRpcClient, Provider},
+};
+
+use crate::config::{Config, EvmRecordVerifier};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum HandlerType {
+    GetDiscordName,
+    GetGithubName,
+    GetTwitterName,
+}
+
+#[derive(Deserialize, Debug)]
+struct GithubUser {
+    login: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct DiscordUser {
+    username: String,
+}
+
+impl EvmRecordVerifier {
+    pub async fn execute_handler(&self, config: &Config, id: FieldElement) -> Result<String> {
+        match self.handler {
```

Comment:
Super clean code, loading as Enum & matching on the variant as you did is perfect